controller/index.js

```diff
@@ -13,6 +12,24 @@ var Controller = function (agol, BaseController) {
   var controller = BaseController()
 
   /**
+  * Manages shared logic for any request that needs a host or key
+  *
+  * @param {object} req - the incoming request object
+  * @param {object} res - the outgoing response object
+  * @param {function} next - calls the next route handler
+  */
+  controller.main = function (req, res, next) {
```

Comment:
Controller methods should generally correspond to web requests for specific resources -- `main` is a little vague in terms of describing what it's doing. This is meant strictly as middleware to add the cache key to the request object before any call to `get /agol/:id/*`, right? I think we should give it a more descriptive name (main can easily be mistaken for index).